R/boxly.R

```diff
@@ -115,11 +115,12 @@ boxly <- function(outdata,
   bar$range <- bar$max - bar$min
   bar <- unique.data.frame(bar)
 
-  bar <- crosstalk::SharedData$new(bar, key = ~param, group = "groupdata")
+  shareddata_id <- paste0("shareddata_", sample(1:9999, 1))
```

Comment:
let's replace this by `basename(tempdir())`.  The current logic did not provide a large randomization space to avoid duplicated name.

Reply:
Using `tempdir()` will generate the same dir name in a session. Should we create a function to generate a random string?

Reply:
Creating a small function is a good idea. Maybe you do not need randomization. The goal can be achieved by using environment variables like below. 

```
sharedata_id <- function(){
  x <- Sys.getenv("boxly_sharedata_id")
  if(x == ""){
    x <- 1
  }else{
    x <- as.numeric(x) + 1
  }
  Sys.setenv(boxly_sharedata_id = x)
  x
} 
```

Reply:
It looks like you need either the uuid package, or the custom unique ID solution in Shiny: https://github.com/ramnathv/htmlwidgets/pull/213. See the latest implementation in `shiny:::createUniqueId()` and `shiny:::withPrivateSeed()`.

I would avoid `Sys.getenv()` and `Sys.setenv()` as it's not worth it to pollute the environment variables for this. Plus, environment variables are unlikely a reliable mechanism (thinking about permission and everything that could affect them).

Reply:
I used uuid and it works. Please take a look

Reply:
Nice! Thanks for the update.